Earn: Add tab for Ads

edanzer opened the pull request:
## Proposed Changes

This PR is for feedback. It adds an 'Ads' tab to Earn page. As is, this tab is only accessible via a card at the bottom. Since we're moving everything else to tab navigation, I wonder if we should move this to a tab as well? This is what it would look like. 

**Without Premium Plan**
<img width="1069" alt="ad-earnings-1" src="https://github.com/Automattic/wp-calypso/assets/21228350/f0020245-dd0e-4d6e-9078-511e499de32d">


**With Premium Plan, but Without Signup**
<img width="1063" alt="ad-earnings-2" src="https://github.com/Automattic/wp-calypso/assets/21228350/4bda25cc-224a-4903-90c7-9184a3059aa1">

**With Premium Plan and Signup**
<img width="1059" alt="ad-earnings-3" src="https://github.com/Automattic/wp-calypso/assets/21228350/9c7f7756-f8ed-4471-9658-ef3b64d494fd">

## Testing Instructions

1) Checkout this branch and go to http://calypso.localhost:3000/earn/YOURDOMAIN. Confirm you see the new Ads tab at the top. 
2) Test the ads tab in all the three states above (not a premium plan, premium plan but not applied, premium plan + applied and accepted) an confirm the screen looks like screenshots above. 

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

edanzer committed:
Add Earn > Ads tab

crisbusquets commented:
I like this idea @edanzer 💡

A few notes:

### General style
The `Ads Dashboard` title feels odd, what do you think about removing the underline and adjusting the padding?

![underline](https://github.com/Automattic/wp-calypso/assets/25607244/13bef06d-bccd-492b-af3d-a9d9934bda50)


In this example:
* `earn__ads-header` only has a margin-top of 20px
* `earn__ads-header h2.formatted-header__title` (h2) has a margin-bottom of 12px

From a type hierarchy point of view, it would be great if the font-size is smaller than the `Earn` title itself, but it's not a blocker :)

### The Back button
I couldn't replicate this. Where does this `Back` button navigate to?

![back-button](https://github.com/Automattic/wp-calypso/assets/25607244/0ec428d0-9408-4355-95d1-5ff1b374089e)



### When Ads are enable
Would it be possible that the `Ads` tab is active when the user navigates to `Payments` or `Settings` (see screenshot below):

![earn-ads-screenshots](https://github.com/Automattic/wp-calypso/assets/25607244/7f821fa6-5f86-41ec-b0b1-356c1e52850f)


### When Ads aren't enabled
Can we add a link in the card/banner where we explain more about Ads? I'd just use the same link we have in the card inside Tools (https://wordads.co/). Just in case the user doesn't read the one there.

Quick mockup:
<img width="1108" alt="Screenshot 2023-09-19 at 17 56 02" src="https://github.com/Automattic/wp-calypso/assets/25607244/5945762f-117a-404b-a47b-0c130823d26e">


Thank you!

edanzer commented:
Hey Cris - long time before I've been able to circle back around on this. In response to your comments above, I will plan to: 
- Add the new Ads tab as proposed in this PR
- Ensure that tab is underlined/highlighted regardless of which sub-tab is active
- Adjust the font-size / padding / underline for the Ads Dashboard secondary title
- Add the explainer link in the card when ads are not enabled
- The Back button section is already removed

edanzer committed:
Update styling for Ads nav

edanzer commented:
@crisbusquets - I've updated this with your changes above. 
  - For Ads Dashboard title, I reduced font, removed underline, and reduced padding
  - Ensured main Ads tab shows active when any sub-page of Ads is active
 
I also tried on my own to make a few styling changes to minimize the appearance of the second set of tabs just for Ads. 

Updated version now looks like this when ads are active:
<img width="1131" alt="monetize-ads-nav" src="https://github.com/Automattic/wp-calypso/assets/21228350/255b03c8-14b0-4031-97e1-ebdd60c59ee4">

And like this when ads are not: 
<img width="1096" alt="monetize-ads-banner" src="https://github.com/Automattic/wp-calypso/assets/21228350/855e189d-6fc8-49cd-9a32-9557d6ce0510">

edanzer committed:
Add link to wordads

edanzer commented:
Going to merge this because it's improvement to what is now a very poor interface, on the understanding we'll have more discussion / design updates / tweaks in the coming days/weeks.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9776069

Thank you @edanzer for including a screenshot in the description! This is really helpful for our translators.

simison commented:
Design convo at p1701247085929269-slack-C9EJ7KSGH

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9776069).
